platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/collections/AbstractLongList.java

```diff
@@ -0,0 +1,614 @@
+        final int firstValidChunkWithBuffer =
+                (int) Math.max((newMinValidIndex - reservedBufferLength) / numLongsPerChunk, 0);
+        final int firstChunkIndexToDelete = firstValidChunkWithBuffer - 1;
+        for (int i = firstChunkIndexToDelete; i >= 0; i--) {
+            final C chunk = chunkList.get(i);
+            if (chunk != null && chunkList.compareAndSet(i, chunk, null)) {
+                releaseChunk(chunk);
+            }
+        }
+
+        // clean up a chunk with data
+        final int firstChunkWithData = (int) (newMinValidIndex / numLongsPerChunk);
+        final long numberOfElementsToCleanUp = (newMinValidIndex % numLongsPerChunk);
+        partialChunkCleanup(firstChunkWithData, numberOfElementsToCleanUp);
+
+        // clean up chunk(s) reserved for buffer
+        for (int i = firstValidChunkWithBuffer; i < firstChunkWithData; i++) {
+            partialChunkCleanup(i, numLongsPerChunk);
+        }
+    }
+
+    /**
+     * Releases a chunk that is no longer in use. Some implementation may preserve the chunk for further use.
+     * @param chunk chunk to release
+     */
+    protected abstract void releaseChunk(C chunk);
+
+    /**
+     * Zeroes out a part of a chunk
+     * @param chunkIndex index of the chunk to clean up
+     * @param entriesToCleanUp number of entries to clean up in the chunk counting from the left boundary
+     */
+    protected abstract void partialChunkCleanup(final int chunkIndex, final long entriesToCleanUp);
+
+    /**
+     * @return a new chunk
+     */
+    protected abstract C createChunk();
+
+    /**
+     * @param totalNumberOfElements total number of elements in the list
+     * @return number of memory chunks that this list may have
+     */
+    protected int calculateNumberOfChunks(final long totalNumberOfElements) {
+        return (int) ((totalNumberOfElements - 1) / numLongsPerChunk + 1);
+    }
+
+    /**
+     * Checks if the value may be put into a LongList at a certain index, given a max capacity.
+     *
+     * @param index the index to check
+     * @param value the value to check
+     * @throws IllegalArgumentException  if the value is impermissible
+     * @throws IndexOutOfBoundsException if the index is out-of-bounds
+     */
+    protected final void checkValueAndIndex(final long index, final long value) {
+        if (index < 0 || index >= maxLongs) {
+            throw new IndexOutOfBoundsException("Index " + index + " is out-of-bounds given capacity " + maxLongs);
+        }
+        if (value == IMPERMISSIBLE_VALUE) {
+            throw new IllegalArgumentException("Cannot put " + IMPERMISSIBLE_VALUE + " into a LongList");
+        }
+    }
+
+    /**
+     * Current max valid index in the list. By default, it's equal to the size of the list, but some
+     * implementations may provide more fine-grained values.
+     *
+     * @return max valid index
+     */
+    protected long getCurrentMax() {
```

Comment:
This method now seems redundant. All the implementations will return `size()` here, right?

Reply:
Yes. It is redundant. I'll remove it